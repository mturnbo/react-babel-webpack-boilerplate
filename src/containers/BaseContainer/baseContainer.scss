.container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;

  -webkit-border-radius: 0px 20px 0px 20px;
  -moz-border-radius: 0px 20px 0px 20px;
  border-radius: 0px 20px 0px 20px;
  border:2px solid #333333;
  background-color:#FEFEFE;
  -webkit-box-shadow: #B3B3B3 10px 10px 10px;
  -moz-box-shadow: #B3B3B3 10px 10px 10px;
  box-shadow: #B3B3B3 10px 10px 10px;
  margin: 20px;

}