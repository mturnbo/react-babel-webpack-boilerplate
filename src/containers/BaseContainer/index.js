export { default } from './BaseContainer.jsx';
