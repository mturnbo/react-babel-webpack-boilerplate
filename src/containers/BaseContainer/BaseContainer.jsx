import React from 'react';
import Base from '../../components/Base';
import data from '../../data/data.json';
import './baseContainer.scss';

class BaseContainer extends React.Component {
  render() {
    const baseList = data.map(entry =>
      <Base title={entry.title} content={entry.content} />
    );

    return (
      <div>
        <h2>Base Container</h2>
        <div className="container">
          {baseList}
        </div>
      </div>
    );
  }
}

export default BaseContainer;
