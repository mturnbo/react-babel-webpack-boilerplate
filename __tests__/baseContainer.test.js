import React from 'react';
import BaseContainer from '../src/containers/BaseContainer';
import { shallow, mount } from 'enzyme';

const wrapper = mount(<BaseContainer />);

test('Base Container Renders', () => {
  expect(wrapper).toMatchSnapshot();
});
