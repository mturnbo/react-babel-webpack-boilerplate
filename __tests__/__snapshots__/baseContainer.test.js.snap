// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Container Renders 1`] = `
<BaseContainer>
  <div>
    <h2>
      Base Container
    </h2>
    <div
      className="container"
    >
      <Base
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        title="Widget"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Widget
          </div>
          <div
            className="base-content"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </div>
        </div>
      </Base>
      <Base
        content="Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt."
        title="Thingee"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Thingee
          </div>
          <div
            className="base-content"
          >
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
          </div>
        </div>
      </Base>
      <Base
        content="Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae."
        title="Whatchamacallit"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Whatchamacallit
          </div>
          <div
            className="base-content"
          >
            Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae.
          </div>
        </div>
      </Base>
      <Base
        content="At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga."
        title="Doohickey"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Doohickey
          </div>
          <div
            className="base-content"
          >
            At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.
          </div>
        </div>
      </Base>
      <Base
        content="Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur."
        title="Thingamajig"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Thingamajig
          </div>
          <div
            className="base-content"
          >
            Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur.
          </div>
        </div>
      </Base>
      <Base
        content="Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?"
        title="Doodad"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Doodad
          </div>
          <div
            className="base-content"
          >
            Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?
          </div>
        </div>
      </Base>
      <Base
        content="Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt."
        title="Thingamabob"
      >
        <div
          className="base"
        >
          <div
            className="base-header"
          >
            Thingamabob
          </div>
          <div
            className="base-content"
          >
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
          </div>
        </div>
      </Base>
    </div>
  </div>
</BaseContainer>
`;
